src/data/media-player.ts

```diff
@@ -210,7 +210,10 @@ export const getCurrentProgress = (stateObj: MediaPlayerEntity): number => {
     (Date.now() -
       new Date(stateObj.attributes.media_position_updated_at!).getTime()) /
     1000.0;
-  return progress;
+  // Prevent negative values, so we do not go back to 59:59 at the start
+  // for example if there are slight clock sync deltas between backend and frontend and
+  // therefore media_position_updated_at might be slightly larger than Date.now().
+  return progress < 0 ? 0 : progress;
```

Comment:
This doesn't actually fix the problem but just hides it for a second.  The elapsed time will still always be wrong.

Can we not correct the time shift by either querying the elapsed time from core, or storing the start time with `performance.now()` and taking the delta from that instead?

Reply:
You're right, but I don't know if the juice is worth the squeeze.

Reply:
Touche.  Admittedly I'm not at all familiar with the details of the media player code to say how difficult the correction would be.

For the back pocket I guess, but a less probable scenario is the frontend clock being ahead, in which case there is no such easy fix except to measure with the same clock.